### DISCUSSION TITLE
How do I get serialization settings used by the output formatter?

### DISCUSSION THREAD

**Discussion by voroninp** [dotnet/aspnetcore#52518] [Q&A]
I am injecting `IOptions<JsonSerializerOptions>` into controller, but I when I serialize I get PascalCased payload whereas endpoints return camelCase.

**Comment by gfoidl** [comment]
@voroninp can you provide a bit more context or a simple self-contained example? So it's hard to give a helpful advice.

Are you aware of [Configure `System.Text.Json`-based formatters](https://learn.microsoft.com/en-us/aspnet/core/web-api/advanced/formatting?view=aspnetcore-8.0#configure-systemtextjson-based-formatters-2)?

**Reply by voroninp** [reply]
I was playing with WebSockets, and wanted to send a payload serialized with the same settings as regular endpoints use, that is with the settings used by Json-based formatters.

But it appears that `IOptions<JsonSerializerOptions>` my controller receives as a dependency holds different settings.

**Reply by voroninp** [reply]
Here's the action:

```
[Route("/api/notifications/reservations")]
[HttpConnect]
[HttpGet]
[ApiExplorerSettings(IgnoreApi = true)]
public async Task<IActionResult> GetReservationUpdates()
{
    if (!HttpContext.WebSockets.IsWebSocketRequest)
    {
        return Problem("Web socket connection is required", statusCode: StatusCodes.Status400BadRequest);
    }

    var tcs = new TaskCompletionSource();
    var cts = new CancellationTokenSource();
    cts.Token.Register(tcs.SetResult);

    using var webSocket = await HttpContext.WebSockets.AcceptWebSocketAsync();
    _reservationsManager.SubscribeForReservationUpdates(async reservation =>
    {
        if (webSocket.State == System.Net.WebSockets.WebSocketState.Open)
        {
            var json = JsonSerializer.SerializeToUtf8Bytes(reservation, _serializerOptions.Value);

            await webSocket.SendAsync(
                json.AsMemory(),
                System.Net.WebSockets.WebSocketMessageType.Text, endOfMessage: true, cts.Token);
        }
        else
        {
            cts.Cancel();
        }
    },
    cts.Token);

    await tcs.Task;

    return Ok();
}
```
